core/chaincode/metrics.go

```diff
@@ -10,54 +10,61 @@ import "github.com/hyperledger/fabric/common/metrics"
 
 var (
 	launchDuration = metrics.HistogramOpts{
-		Namespace:    "chaincode",
-		Name:         "launch_duration",
-		Help:         "The time to launch a chaincode.",
-		LabelNames:   []string{"chaincode", "success"},
-		StatsdFormat: "%{#fqname}.%{chaincode}.%{success}",
+		Namespace:       "chaincode",
+		Name:            "launch_duration",
+		Help:            "The time to launch a chaincode.",
+		LabelNames:      []string{"chaincode", "success"},
+		StatsdFormat:    "%{#fqname}.%{chaincode}.%{success}",
+		SourceComponent: "chaincode",
```

Comment:
Here we have a metric that isn't "peer" or "orderer" as documented where `SourceComponent` was defined.

It's also confusing as this is emitted by the peer for chaincode that it has launched.